include/rang.hpp

```diff
@@ -117,8 +117,9 @@ namespace rang {
 	template <typename T>
 	inline enable<T> operator<<(std::ostream &os, T const value)
 	{
+        static bool hasColors = supportsColor();
```

Comment:
think of this too, would this be generated seperately in all 3 function generated by this template?